lib/phoenix_component/declarative.ex

```diff
@@ -1111,7 +1116,7 @@ defmodule Phoenix.Component.Declarative do
       end)
 
     for {name, {line, _column, _type_value}} <- attrs,
-        !(global_attr && __global__?(caller_module, Atom.to_string(name), global_attr)) do
+        !global_attr || (global_attr && Keyword.get(global_attr.opts, :verify, true) && !__global__?(caller_module, Atom.to_string(name), global_attr)) do
```

Comment:
Should we move the verify check to inside `__global__?`, so the logic here is not as complex?